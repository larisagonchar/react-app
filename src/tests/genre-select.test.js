import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import GenreSelect from '../components/genre-select/genre-select';

describe('GenreSelect component', () => {
  const genres = ['All', 'Action', 'Adventure', 'Comedy', 'Crime', 'Drama', 'Family', 'Thriller'];
  const selectedGenre = 'Drama';

  test('render all genres', () => {
    const { asFragment } = render(<GenreSelect genres={genres} />);

    expect(asFragment).toMatchSnapshot();
  });

  test('highlight selected genre', () => {
    render(<GenreSelect genres={genres} selectedGenre={selectedGenre} />);
    const selectedGenreItem = screen.getByText('Drama');

    expect(selectedGenreItem).toHaveClass('genre-select__item_selected');
  });

  test('call onSelect when user select item', () => {
    const onItemClickMock = jest.fn();
    render(<GenreSelect genres={genres} selectedGenre={selectedGenre} onSelect={onItemClickMock} />);

    const genre = screen.getByText('Action');
    userEvent.click(genre);

    expect(onItemClickMock).toHaveBeenCalledWith('Action');
  });
});