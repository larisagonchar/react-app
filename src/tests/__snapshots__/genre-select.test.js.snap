// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreSelect component should render all genres 1`] = `[Function]`;
